modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotRestoreProcess.java

```diff
@@ -0,0 +1,777 @@
+
+                snpMgr.runSnapshotVerfification(metas).listen(
+                    f0 -> {
+                        if (f0.error() != null) {
+                            fut.onDone(f0.error());
+
+                            return;
+                        }
+
+                        IdleVerifyResultV2 res = f0.result();
+
+                        if (!F.isEmpty(res.exceptions()) || res.hasConflicts()) {
+                            StringBuilder sb = new StringBuilder();
+
+                            res.print(sb::append, true);
+
+                            fut.onDone(new IgniteException(sb.toString()));
+
+                            return;
+                        }
+
+                        SnapshotRestorePrepareRequest req = new SnapshotRestorePrepareRequest(UUID.randomUUID(),
+                            snpName, dataNodes, cacheGrpNames, F.first(dataNodes));
+
+                        prepareRestoreProc.start(req.requestId(), req);
+                    }
+                );
+            }
+        );
+
+        return new IgniteFutureImpl<>(fut);
+    }
+
+    /**
+     * Check if snapshot restore process is currently running.
+     *
+     * @return {@code True} if the snapshot restore operation is in progress.
+     */
+    public boolean isRestoring() {
+        return opCtx != null;
+    }
+
+    /**
+     * Check if the cache or group with the specified name is currently being restored from the snapshot.
+     *
+     * @param cacheName Cache name.
+     * @param grpName Cache group name.
+     * @return {@code True} if the cache or group with the specified name is currently being restored.
+     */
+    public boolean isRestoring(String cacheName, @Nullable String grpName) {
+        SnapshotRestoreContext opCtx0 = opCtx;
+
+        if (opCtx0 == null)
+            return false;
+
+        Map<Integer, StoredCacheData> cacheCfgs = opCtx0.cfgs;
+
+        int cacheId = CU.cacheId(cacheName);
+
+        if (cacheCfgs.containsKey(cacheId))
+            return true;
+
+        for (File grpDir : opCtx0.dirs) {
+            String locGrpName = FilePageStoreManager.cacheGroupName(grpDir);
+
+            if (grpName != null) {
+                if (cacheName.equals(locGrpName))
+                    return true;
+
+                if (CU.cacheId(locGrpName) == CU.cacheId(grpName))
+                    return true;
+            }
+            else if (CU.cacheId(locGrpName) == cacheId)
+                return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Finish local cache group restore process.
+     */
+    private void finishProcess() {
+        finishProcess(null);
+    }
+
+    /**
+     * Finish local cache group restore process.
+     *
+     * @param err Error, if any.
+     */
+    private void finishProcess(@Nullable Throwable err) {
+        SnapshotRestoreContext opCtx0 = opCtx;
+
+        if (err != null) {
+            log.error("Failed to restore snapshot cache group" + (opCtx0 == null ? "" :
```

Comment:
Can you please fix the Intellij IDEA suggestions here?

Reply:
Done